fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -362,28 +428,38 @@ private boolean isTruckOrTrailer(VehicleClass vehicleClass) {
                 || SumoVehicleClassMapping.toSumo(vehicleClass).equals("trailer");
     }
 
+    private void propagateRouteIfAbsent(String routeId, VehicleRoute route) throws InternalFederateException {
+        // if the route is already known (because it is defined in a route-file) don't add route
+        if (routeCache.containsKey(routeId)) {
+            log.warn("Couldn't add Route {}, because it is already known to SUMO.", routeId);
+        } else {
+            routeCache.put(routeId, route);
+            traci.getRouteControl().addRoute(routeId, route.getEdgeIdList());
+        }
+    }
+
     /**
-     * Writes a new SUMO route file based on the registered vehicle types.
+     * Writes a new SUMO additional-file based on the registered vehicle types.
      *
      * @param typesInit Interaction contains predefined vehicle types.
      */
     private void writeTypesFromMapping(VehicleTypesInitialization typesInit) {
-        initVehicleTypeRouteFileCreator();
+        initVehicleTypesWriter();
         // stores the rou.xml file to the working directory. this file is required for SUMO to run
-        sumoRouteFileCreator
+        sumoVehicleTypesWriter
                 .addVehicleTypes(typesInit.getTypes())
                 .store();
     }
 
-    private void initVehicleTypeRouteFileCreator() {
+    private void initVehicleTypesWriter() {
         File dir = new File(descriptor.getHost().workingDirectory, descriptor.getId());
         String subDir = new File(sumoConfig.sumoConfigurationFile).getParent();
         if (StringUtils.isNotBlank(subDir)) {
             dir = new File(dir, subDir);
         }
         // keep single instance
-        if (sumoRouteFileCreator == null) {
-            sumoRouteFileCreator = new SumoRouteFileCreator(dir, sumoConfig);
+        if (sumoVehicleTypesWriter == null) {
+            sumoVehicleTypesWriter = new SumoVehicleTypesWriter(dir, sumoConfig);
         }
```

Comment:
do we still need this extra method/field? we only use the SumoVehicleTypesWriter once now, aren't we? we could move this to `writeTypesFromMapping` (and rename the method to `writeTypesFromRti`)

Reply:
true, I missed this